core/src/test/scala/unit/kafka/server/SessionExpireListenerTest.scala

```diff
@@ -0,0 +1,77 @@
+/**
+  * Licensed to the Apache Software Foundation (ASF) under one or more
+  * contributor license agreements.  See the NOTICE file distributed with
+  * this work for additional information regarding copyright ownership.
+  * The ASF licenses this file to You under the Apache License, Version 2.0
+  * (the "License"); you may not use this file except in compliance with
+  * the License.  You may obtain a copy of the License at
+  *
+  *    http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+
+package kafka.server
+
+import kafka.api.ApiVersion
+import kafka.utils.ZkUtils
+import org.I0Itec.zkclient.ZkClient
+import org.apache.kafka.common.metrics.Metrics
+import org.apache.zookeeper.Watcher
+import org.easymock.EasyMock
+import org.junit.{After, Assert, Before, Test}
+import Assert._
+import org.apache.kafka.common.MetricName
+import org.apache.kafka.common.utils.MockTime
+
+class SessionExpireListenerTest {
+
+  private var metrics: Metrics = _
+  private var time: MockTime = _
+  private val brokerId = 1
+
+  @Before
+  def setUp() {
+    time = new MockTime
+    metrics = new Metrics(time)
+  }
+
+  @After
+  def tearDown() {
+    metrics.close()
+  }
+
+  @Test
+  def testSessionExpireListenerMetrics() {
```

Comment:
One way to generate a Expired event is to create a session, close it, and create a new one using the same session id. I'm not sure we can do it with the ZkClient API, though, but here is how to do it with the raw zk client:

```
 ZooKeeper zk = new ZooKeeper("localhost:2181", 10000, this);
 // Wait until connected 
 long sessionId = zk.getSessionId();
 zk.close();
 zk = new ZooKeeper("localhost:2181", 10000, this, sessionId, new byte[0]);
```